File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        if len(nums) < 3:
            return False

        left, mid = float("inf"), float("inf")

        for num in nums:
            if num > mid:
                return True
            elif num < left:
                left = num
            elif left < num < mid:
                mid = num

        return False

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.rs
impl Solution {
    pub fn increasing_triplet(nums: Vec<i32>) -> bool {
        if nums.len() < 3 {
            return false;
        }

        let mut left = i32::MAX;
        let mut mid = i32::MAX;

        for &num in &nums {
            if num > mid {
                return true;
            } else if num < left {
                left = num;
            } else if num > left && num < mid {
                mid = num;
            }
        }

        false
    }
}
